api/py/ai/chronon/repo/compile.py

```diff
@@ -163,13 +163,16 @@ def extract_and_convert(chronon_root, input_path, output_root, debug, force_over
                 chronon_root_path, entity_dependency_tracker, full_output_root, name, obj, obj_class, log_level
             )
 
+            _handle_deprecation_warning(obj, obj_class, new_group_bys, new_joins)
+
             # Update the accumulated dictionaries with the new entries
             extra_dependent_group_bys_to_materialize.update(new_group_bys)
             extra_dependent_joins_to_materialize.update(new_joins)
 
+    dependencies = {}
+    dependencies.update({**extra_dependent_group_bys_to_materialize, **extra_dependent_joins_to_materialize})
```

Comment:
Seems like the `dependencies` variable is only used in the IF block below (L175-L180). What's the reason to remove this out of the IF block?

Reply:
updated